hivemind/optim/power_sgd_averager.py

```diff
@@ -0,0 +1,249 @@
+import asyncio
+import contextlib
+import faulthandler
+import math
+import multiprocessing as mp
+from typing import Any, Iterable, Optional, Sequence
+
+import numpy as np
+import torch
+
+import hivemind
```

Comment:
Better to avoid broad imports inside the library, since it makes refactoring more difficult (due to the possibility of circular imports) and unnecessarily increases the scope of the import statement